docs/source/adr/0001-httpx-async-first-architecture.md

```diff
@@ -0,0 +1,487 @@
+# ADR 0001: HTTPX Async-First Architecture with Thin Sync Wrappers
+
+## Status
+
+**Proposed** - For caldav v3.0
+**Supersedes** - #555
+
+## Context
+
+The caldav library currently uses `requests` (with optional `niquests` support) for HTTP communication. This synchronous-only approach limits the library's usefulness in modern async Python applications. A previous attempt to add async support via parallel implementations created significant code duplication and maintenance burden.
+
+### Current Architecture
+
+The library's HTTP layer is centralized in `DAVClient` (`davclient.py`), which provides:
+- Core HTTP methods: `request()`, `propfind()`, `proppatch()`, `report()`, `mkcalendar()`, `put()`, `post()`, `delete()`, `options()`
+- Session management via `requests.Session()`
+- Authentication handling (Basic, Digest, Bearer)
+- Response parsing via `DAVResponse`
+
+All other classes (`DAVObject`, `Calendar`, `Principal`, `CalendarSet`, `CalendarObjectResource`, `Event`, `Todo`, `Journal`) delegate HTTP operations through `self.client` to `DAVClient`.
+
+### Key Files and Structure
+
+```
+caldav/
+├── davclient.py          # DAVClient, DAVResponse, HTTP layer (~1100 lines)
+├── davobject.py          # DAVObject base class (~430 lines)
+├── collection.py         # Principal, Calendar, CalendarSet (~1300 lines)
+├── calendarobjectresource.py  # Event, Todo, Journal, FreeBusy (~1660 lines)
+├── search.py             # CalDAVSearcher (~510 lines)
+├── requests.py           # HTTPBearerAuth (~20 lines)
```

Comment:
I just had a look at this file yesterday.  The file was introduced by someone else in a pull request as far as I can remember.  I'm considering to move request-handling logic from davclient to requests.